### PR TITLE
resolves #72

### LINKED ISSUE #72: make it that when an election starting and ending date is changed the scheduler is updated as well

### PR DESCRIPTION
**Pull request by AnointingMax**

### CODE REVIEW

**Review by LeanKhan — APPROVED** [review]

File: app/routes.py [hunk 1]
```diff
@@ -140,6 +141,19 @@ def update_election(link):
 
         db.session.commit()
 
+        # the scheduler
+        try:
+            # remove the previous scheduled event then recreate with form data
+            scheduler.remove_job(id="start " + election.name)
```
Comment (LeanKhan) [inline app/routes.py]: Hey, can we use the Election.id instead of Election.name since name can change, but ID cannot?

**Review by danieliheonu — APPROVED** [review]

**Review by danieliheonu — APPROVED** [review]

**Review by danieliheonu — APPROVED** [review]
Nice work 👍

File: app/forms.py [hunk 2]
```diff
@@ -28,6 +30,17 @@ class ElectionForm(FlaskForm):
     password = StringField('Password')
     submit = SubmitField('Submit')
 
+    def validate_starting_at(form, field):
+        if field.data < datetime.utcnow() :
+            raise ValidationError('Starting date must be greater than current time')
```
Comment (LeanKhan) [inline app/forms.py:35]: :+1:

File: app/templates/elections/form.html [hunk 3]
```diff
@@ -2,6 +2,46 @@
...
+        Starting At<br>
+        {{ form.starting_at(class='select_starttime') }}
+        {% for error in form.starting_at.errors %}
+        <span style="color: red;display: block">[{{ error }}]</span>
+        {% endfor %}
+        <!-- you put the loop for errors twice -->
+    </p>
+    <p>
+        Ending At<br>
+        {{ form.ending_at(class='select_endtime') }}
```
Comment (LeanKhan) [inline app/templates/elections/form.html:31]: hmm, looks very different from the current form...

**Pull request merged**